packages/toc/src/toc_item.tsx

```diff
@@ -57,9 +119,77 @@ class TOCItem extends React.Component<IProperties, IState> {
       event.stopPropagation();
       heading.onClick();
     };
+    /**
+     * Callback invoked upon clicking a menu item to run code cells.
+     */
+    const runCells = () => {
+      let code: INotebookHeading[];
+      let h: INotebookHeading;
+      let i: number;
+
+      // Find the heading in the list of ToC headings...
+      const headings = toc;
+      for (i = 0; i < headings.length; i++) {
+        if (heading === headings[i]) {
+          break;
+        }
+      }
+      code = [];
+
+      // Check if the current heading is a "code" heading...
+      h = heading as INotebookHeading;
+      if (h.type === 'code') {
+        code.push(h);
+      }
+      // Find all nested code headings...
+      else {
+        const level = heading.level;
+        for (i = i + 1; i < headings.length; i++) {
+          h = headings[i] as INotebookHeading;
+          if (h.level <= level) {
+            break;
+          }
+          if (h.type === 'code') {
+            code.push(h);
+          }
+        }
+      }
```

Comment:
This appears to have a large overlap with the hasCodeCells, would it make sense to extract it to another function?

Reply:
made the two functions into nestedCodeCells with an option to run code or not